includes/multi-currency/FrontendCurrencies.php

```diff
@@ -270,7 +276,16 @@ public function init_order_currency( $arg ) {
 			return $arg;
 		}
 
+		// We remove the filters here becuase 'wc_get_order' triggers the 'wc_get_price_decimal_separator' filter.
+		remove_filter( 'wc_get_price_decimal_separator', [ $this, 'get_price_decimal_separator' ], 900 );
+		remove_filter( 'wc_get_price_decimal_separator', [ $this, 'init_order_currency_from_query_vars' ], 900 );
 		$order = ! $arg instanceof WC_Order ? wc_get_order( $arg ) : $arg;
+		// Note: it's important that 'init_order_currency_from_query_vars' is called before
+		// 'get_price_decimal_separator' because the order currency is often required to
+		// determine the decimal separator. That's why the priority on 'init_order_currency_from_query_vars'
+		// is explicity lower than the priority of 'get_price_decimal_separator'.
+		add_filter( 'wc_get_price_decimal_separator', [ $this, 'init_order_currency_from_query_vars' ], 900 );
+		add_filter( 'wc_get_price_decimal_separator', [ $this, 'get_price_decimal_separator' ], 900 );
```

Comment:
Just noticed this after approving, but according to your notes, the priority on `get_price_decimal_separator` should be `901` here as well.

Reply:
Fixed in https://github.com/Automattic/woocommerce-payments/pull/8387/commits/3f93b1fdc324bfd7a1029265635e63c7afd5f42b